portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,39 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+// photos data
+import photos_data from './photos-data.js';
+
 /**
- * Adds a random greeting to the page.
+ * Create photo component with caption 
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
-
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+function createPhoto(photo) {
+    let component = [
+        '<div class="col-4">',
+            '<figure class="figure">',
+                '<img src=' + photo.src + ' class="figure-img img-fluid rounded" alt=' + photo.location + '>',
+                '<div class="row figure-caption-container">',
+                    '<figcaption class="figure-caption photo-location">' + photo.location + '</figcaption>',
+                    '<span class="dot"></span>',
+                    '<figcaption class="figure-caption photo-date">' + photo.date + '</figcaption>',
+                '</div>',
+            '</figure>',
+        '</div>'
+    ];
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+  return $(component.join(''));
 }
+
+/**
+ * Map photos into gallery
+ */
+ function mapPhotos() {
+     photos_data.forEach(
```

Comment:
This isn't in the style guide I can link to you, but it's preferable to use standard for of loops. This would be:

```js
for ( const photo of photos_data) {
  $('#gallery').append(createPhoto(photo));
}
```